public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -100,6 +103,41 @@ export function getRuleHealth(health: string): RuleHealth | undefined {
   }
 }
 
+export interface RuleOrigin {
+  pluginId: string;
+}
+
+export function getRuleOrigin(rule: CombinedRule): RuleOrigin | undefined {
+  // com.grafana.origin=plugin/<plugin-identifier>
+  // Prom and Mimir do not support dots in label names 😔
+  const origin = rule.labels[GRAFANA_ORIGIN_LABEL];
+  if (!origin) {
+    return undefined;
+  }
+
+  const match = origin.match(/^plugin\/(?<pluginId>.+)$/);
+  if (!match?.groups) {
+    return undefined;
+  }
+
+  const pluginId = match.groups['pluginId'];
+  const pluginInstalled = isPluginInstalled(pluginId);
+
+  if (!pluginInstalled) {
+    return undefined;
+  }
+
+  return { pluginId };
+}
+
+function isPluginInstalled(pluginId: string) {
+  return Boolean(config.apps[pluginId]);
+}
+
+export function isPluginProvidedRule(rule: CombinedRule): boolean {
+  return Boolean(getRuleOrigin(rule));
```

Comment:
Would you not want to check for existence of `pluginId` on the origin? Rather than generally "We found something as an origin"?

Reply:
`getRuleOrigin` already checks if the plugin exists

Reply:
I'll add tests covering that